src/transformStateWithClones.js

```diff
@@ -6,8 +6,35 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(initialState, actions) {
+  const stateHistory = [];
+  let currentState = { ...initialState };
+
+  actions.forEach((action) => {
+    let stateCopy = { ...currentState };
+
+    switch (action.type) {
+      case 'clear':
+        stateCopy = {};
+        break;
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...action.extraData };
```

Comment:
Consider checking if `action.extraData` is defined before attempting to spread it into `stateCopy`. This will prevent potential runtime errors if `extraData` is undefined.